File: Problems/14/main.go
package main

func main() {}

func longestCommonPrefix(strs []string) string {
	check := func(strs []string, idx int) bool {
		if len(strs[0]) == idx {
			return false
		}
		tmp := strs[0][idx]
		for i := 1; i < len(strs); i++ {
			if idx == len(strs[i]) || strs[i][idx] != tmp {
				return false
			}
		}
		return true
	}

	res := 0
	for ; res < 201; res++ {
		if !check(strs, res) {
			break
		}
	}
	return strs[0][:res]
}

File: Problems/14/main.cpp
#include <vector>
#include <iostream>
#include <string>
#include <map>
#include <stack>
#include <algorithm>
#include <numeric>
#include <unordered_set>
#include <unordered_map>
#include <queue>
#include <functional>
#include <sstream>
#include <set>
#include <cctype>

using namespace std;

class Solution
{
public:
    string longestCommonPrefix(vector<string> &strs)
    {
        int n = strs.size();
        int p = 0;
        while (true)
        {
            bool ok = true;
            if (p >= strs[0].size())
                break;
            for (int i = 1; i < n; i++)
            {
                if (p >= strs[i].size() || strs[i][p] != strs[i - 1][p])
                {
                    ok = false;
                    break;
                }
            }
            if (ok)
                p++;
            else
                break;
        }
        return strs[0].substr(0, p);
    }
};
